internal/buffer/settings.go

```diff
@@ -20,7 +20,19 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 	} else if option == "statusline" {
 		screen.Redraw()
 	} else if option == "filetype" {
-		config.InitLocalSettings(b.Settings, b.Path)
+		settings := config.ParsedSettings()
+		settings["filetype"] = nativeValue
+		config.InitLocalSettings(settings, b.Path)
+		for k, v := range config.DefaultCommonSettings() {
+			if k == "filetype" {
+				continue
+			}
+			if _, ok := settings[k]; ok {
+				b.SetOptionNative(k, settings[k])
+			} else {
+				b.SetOptionNative(k, v)
+			}
+		}
```

Comment:
Hmm... this brings back the original issue reported in #3342: setting `filetype` overrides volatile settings (even though for the current buffer only)?